src/components/elements/payment-carousel.js

```diff
@@ -0,0 +1,118 @@
+import IWallet from 'images/svg/payment-i-wallet.svg'
+import Paytrust from 'images/svg/payment-paytrust.svg'
+import Jeton from 'images/svg/payment-jeton.svg'
+
+const PaymentSection = styled(SectionContainer)`
+    padding: 2.4rem 0;
+`
+const CarouselContainer = styled(Container)`
+    overflow: hidden;
+    justify-content: space-between;
+
+    div {
+        transition: transform ${props => (props.transition ? '1s' : '0')};
+        transform: translate(${props => props.position}rem);
+    }
+`
+const PaymentWrapper = styled.div`
+    max-width: 9.6rem;
+    margin-right: 3.2rem;
+`
+
+class PaymentCarousel extends React.Component {
+    intervalRef = undefined
+    timeoutRef = undefined
+    MyRef = React.createRef()
+    state = {
+        paymentArray: [
+            { key: 0, Name: Mastercard },
+            { key: 1, Name: Visa },
+            { key: 2, Name: Paytrust },
+            { key: 3, Name: Banktransfer },
+            { key: 4, Name: Netller },
+            { key: 5, Name: Skrill },
+            { key: 6, Name: IWallet },
+            { key: 7, Name: Jeton },
+            { key: 8, Name: InternetBankTransfer },
+            { key: 9, Name: WebMoney },
+            { key: 10, Name: Qiwi },
+            { key: 11, Name: Yandex },
+            { key: 12, Name: PerfectMoney },
+            { key: 13, Name: Maestro },
+            { key: 14, Name: Sticpay },
+            { key: 15, Name: Fasapay },
+            { key: 16, Name: Payscale },
+        ],
+        position: 0,
+        transition: true,
+    }
+    handler = entries => {
+        let entry
+
+        for (entry of entries) {
+            if (entry.isIntersecting) {
+                this.intervalRef = window.setInterval(this.handleInterval, 1700)
+            } else {
+                window.clearTimeout(this.timeoutRef)
+                window.clearInterval(this.intervalRef)
+            }
+        }
+    }
+    handleInterval = () => {
+        this.setState({
+            transition: true,
+            position: this.state.position - 12.8,
+        })
+        this.timeoutRef = setTimeout(() => {
+            const newPaymentArray = this.state.paymentArray
+            const newPaymentIcon = newPaymentArray.shift()
+            newPaymentArray.push(newPaymentIcon)
+            this.setState({
+                transition: false,
+                paymentArray: newPaymentArray,
+                position: 0,
+            })
+        }, 1100)
+    }
+    componentDidMount() {
+        const node = this.MyRef.current
+        let observer = new IntersectionObserver(this.handler)
+        observer.observe(node)
+    }
+    render() {
+        return (
+            <PaymentSection>
+                <CarouselContainer
+                    ref={this.MyRef}
+                    transition={this.state.transition}
+                    position={this.state.position}
+                >
+                    {this.state.paymentArray.map(Payment => (
```

Comment:
Why not decouple the payment and the carousel?

Reply:
Because `payment` is carousel by itself. So I think it's better to be in the same place.
Also, we haven't any other carousel. if we have another one, we can decouple it and make it reusable.

Reply:
I think we should decouple them now it will be easier to maintain and more extensible

Reply:
Ok